src/schema/field.rs

```diff
@@ -3,12 +3,11 @@ use std::io;
 use std::io::Read;
 use std::io::Write;
 
-/// `Field` is actually a `u8` identifying a `Field`
-/// The schema is in charge of holding mapping between field names
-/// to `Field` objects.
+/// `Field` is represented by an unsigned 32-bit integer type
+/// The schema holds the mapping between field names and `Field` objects.
 ///
-/// Because the field id is a `u8`, tantivy can only have at most `255` fields.
-/// Value 255 is reserved.
+/// Because the field id is a `u32`, tantivy can have at most `4,294,967,295` fields.
+/// The last value is reserved.
```

Comment:
It is valid to use `Field(u32::max_value())` as a way to represent an undefined field.